cli/src/commands/fix.rs

```diff
@@ -250,10 +253,11 @@ fn run_tool(
     file_to_fix: &FileToFix,
     old_content: &[u8],
 ) -> Result<Vec<u8>, ()> {
-    // TODO: Pipe stderr so we can tell the user which commit, file, and tool it is
-    // associated with.
     let mut vars: HashMap<&str, &str> = HashMap::new();
     vars.insert("path", file_to_fix.repo_path.as_internal_file_string());
+    if let Some(root) = workspace_root.to_str() {
+        vars.insert("root", root);
+    }
```

Comment:
nit: better to return error if `.to_str()` failed?

Reply:
The repo could be in a path that isn't UTF-8, which is when `to_str()` would return `None`.

If we return an error, then `jj fix` doesn't work at all even if the fix tool doesn't use `$root` in its `command`. That's why I thought it would be okay to ignore the failure case here.

Reply:
A warning or error would be nice if running `jj fix`, the fix command actually uses `$root`, and `workspace_root.to_str()` returned `None`.

But I'd only want to emit that once, not once per file, so it seems slightly tricky unless `Ui` already has a way to deduplicate warnings.

Reply:
Oh I posted my last comment without seeing your reply. Yes, I'll add the TODO, thanks.